x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/SimpleRole.java

```diff
@@ -155,4 +163,38 @@ public int hashCode() {
         return result;
     }
 
+    public static final Setting<Integer> CACHE_SIZE_SETTING = Setting.intSetting(
+        "xpack.security.authz.store.role.has_privileges.cache.max_size",
+        1000,
+        Setting.Property.NodeScope
+    );
```

Comment:
Nit: I noticed the other `xpack.security.authz.store.*` settings use "roles" (plural) instead of "role". I think it would be more consistent that instead. Also, maybe you could move this declaration to the top of the class.

Reply:
As commented, the cache is per `role`. So I intentionally changed `roles` to `role`. But maybe it is more confusing than helpful. I changed it back to be `role` again. Also moved the line to the top.